storm-core/src/jvm/org/apache/storm/daemon/supervisor/timer/RunProfilerActions.java

```diff
@@ -0,0 +1,214 @@
+        public Object call(int exitCode) {
+            LOG.info("{} profile-action exited for {}", logPrefix, exitCode);
+            try {
+                if (stop)
+                    stormClusterState.deleteTopologyProfileRequests(stormId, profileRequest);
+            } catch (Exception e) {
+                LOG.warn("failed delete profileRequest: " + profileRequest);
+            }
+            return null;
+        }
+    }
+
+    public RunProfilerActions(SupervisorData supervisorData) {
+        this.conf = supervisorData.getConf();
+        this.stormClusterState = supervisorData.getStormClusterState();
+        this.hostName = supervisorData.getHostName();
+        this.profileCmd = (String) (conf.get(Config.WORKER_PROFILER_COMMAND));
+        this.supervisorData = supervisorData;
+    }
+
+    @Override
+    public void run() {
+        Map<String, List<ProfileRequest>> stormIdToActions = supervisorData.getStormIdToProfilerActions().get();
+        try {
+            for (Map.Entry<String, List<ProfileRequest>> entry : stormIdToActions.entrySet()) {
+                String stormId = entry.getKey();
+                List<ProfileRequest> requests = entry.getValue();
+                if (requests != null) {
+                    for (ProfileRequest profileRequest : requests) {
+                        if (profileRequest.get_nodeInfo().get_node().equals(hostName)) {
+                            boolean stop = System.currentTimeMillis() > profileRequest.get_time_stamp();
+                            Long port = profileRequest.get_nodeInfo().get_port().iterator().next();
+                            String targetDir = ConfigUtils.workerArtifactsRoot(conf, String.valueOf(port));
+                            Map stormConf = ConfigUtils.readSupervisorStormConf(conf, stormId);
+
+                            String user = null;
+                            if (stormConf.get(Config.TOPOLOGY_SUBMITTER_USER) != null) {
+                                user = (String) (stormConf.get(Config.TOPOLOGY_SUBMITTER_USER));
+                            }
+                            Map<String, String> env = null;
+                            if (stormConf.get(Config.TOPOLOGY_ENVIRONMENT) != null) {
+                                env = (Map<String, String>) stormConf.get(Config.TOPOLOGY_ENVIRONMENT);
+                            } else {
+                                env = new HashMap<String, String>();
+                            }
+
+                            String str = ConfigUtils.workerArtifactsPidPath(conf, stormId, port.intValue());
+                            StringBuilder stringBuilder = new StringBuilder();
+
+                            try (FileReader reader = new FileReader(str);
+                                 BufferedReader br = new BufferedReader(reader)) {
+                                int c;
+                                while ((c = br.read()) >= 0) {
+                                    stringBuilder.append(c);
```

Comment:
Appending an int is not what we want, we want to append an ASCII char.  We really should be doing a readLine.

```
                            String pidFile = ConfigUtils.workerArtifactsPidPath(conf, stormId, port.intValue());
                            String workerPid = null;

                            try (FileReader reader = new FileReader(str);
                                 BufferedReader br = new BufferedReader(reader)) {
                                 workerPid = br.readLine().trim();
                            }
```